### ISSUE TITLE
Get grid_gap to work when RowSideColors is used

### ISSUE THREAD

**Issue by talgalili** [talgalili/heatmaply#53]
For example, in here:

```r
library(heatmaply)
heatmaply(iris, grid_gap = 3)
```

We also want the grid to work on the row side color

![image](https://cloud.githubusercontent.com/assets/976006/24898730/81ef685a-1ea6-11e7-86ce-2d8a1736204e.png)

**Comment by Alanocallaghan** [comment]
You could try replacing `geom_raster` with `geom_tile` in `side_color_plot`, as that should be the only major difference.

**Comment by talgalili** [comment]
I'll try it now.

**Comment by talgalili** [comment]
Nope, doesn't seem to work. I pushed this attempt to the main repo.
Yet I get:

```r
> heatmaply(iris, grid_gap = 2)
Warning message:
'scatter' objects don't have these attributes: 'xgap'
Valid attributes include:
'type', 'visible', 'showlegend', 'legendgroup', 'opacity', 'name', 'uid', 'hoverinfo', 'stream', 'x', 'x0', 'dx', 'y', 'y0', 'customdata', 'dy', 'ids', 'text', 'mode', 'hoveron', 'line', 'connectgaps', 'fill', 'fillcolor', 'marker', 'textposition', 'textfont', 'r', 't', 'error_y', 'error_x', 'xaxis', 'yaxis', 'xcalendar', 'ycalendar', 'xsrc', 'ysrc', 'customdatasrc', 'idssrc', 'textsrc', 'textpositionsrc', 'rsrc', 'tsrc', 'key', 'set', 'frame', 'transforms', '_isNestedKey', '_isSimpleKey'
 
```

With this output:

![image](https://cloud.githubusercontent.com/assets/976006/24901492/e7f3fcac-1eaf-11e7-93a6-42acc1c9e58d.png)

**Comment by Alanocallaghan** [comment]
Depends on #38 as xgap only works for heatmaps, and ggplotly only converts to heatmaps when numeric (I think)

**Comment by talgalili** [comment]
Would this be worth opening an issue in ropensci/plotly? (I suspect not, but wanted to check in with you also)

**Comment by Alanocallaghan** [comment]
No, I think it's pretty fundamental to how plotly.js handles heatmaps. It would be nice to get categorical heatmap support in plotly.js but I suspect that's a big ask! BTW, I'm making progress on that, but mouseover text may be a problem... Not sure if it's worth sacrificing that for the improvements that would be made

**Comment by talgalili** [comment]
Mouseover is essential I believe. It might be worth suggesting as an issue on ropensci/plotly (I'll upvote it :) )

**Comment by Alanocallaghan** [comment]
I was just passing hovertext in the wrong format! Should be up and running with this in a few days

**Comment by talgalili** [comment]
That's great to hear, thanks!

**Comment by Alanocallaghan** [comment]
Works in plot_method = "plotly", still broken in ggplotly.
